src/ProductConstructionService/ProductConstructionService.Api/Queue/JobConsumer.cs

```diff
@@ -8,22 +8,19 @@
 
 namespace ProductConstructionService.Api.Queue;
 
-public class JobProcessor(
-    ILogger<JobProcessor> logger,
-    IOptions<JobProcessorOptions> options,
-    JobProcessorScopeManager scopeManager,
+public class JobConsumer(
+    ILogger<JobConsumer> logger,
+    IOptions<JobConsumerOptions> options,
+    JobScopeManager scopeManager,
     QueueServiceClient queueServiceClient)
     : BackgroundService
 {
-    private readonly ILogger<JobProcessor> _logger = logger;
-    private readonly IOptions<JobProcessorOptions> _options = options;
-    private readonly JobProcessorScopeManager _scopeManager = scopeManager;
+    private readonly ILogger<JobConsumer> _logger = logger;
+    private readonly IOptions<JobConsumerOptions> _options = options;
+    private readonly JobScopeManager _scopeManager = scopeManager;
 
     protected override async Task ExecuteAsync(CancellationToken cancellationToken)
     {
-        // The API won't be initialized until the BackgroundService goes async. Since the scopeManagers blocks aren't async, we have to do it here
-        await Task.Delay(1000);
```

Comment:
@dkurepa this seems wrong. I removed it and everything works. How did this manifest?

Reply:
The API doesn't initialize until the BackgroundService awaits something. Everything worked for you because when running locally, we start the service in the `Working` state, so it doesn't block. But when running on Azure, the API won't initialize, because we'll block on the `ManualEventReset`, and not have the API to let it through

Reply:
But why not yield the thread only?

Reply:
you mean only when running in cloud? I thought this looked a bit cleaner, you don't really feel the 1 sec delay when running the service